docs/providers/aws/events/kafka.md

```diff
@@ -0,0 +1,104 @@
+<!--
+title: Serverless Framework - AWS Lambda Events - Self-managed Apache Kafka
+menuText: Kafka
+description:  Setting up AWS self-managed Apache Kafka Events with AWS Lambda via the Serverless Framework
+layout: Doc
+-->
+
+<!-- DOCS-SITE-LINK:START automatically generated  -->
+
+### [Read this on the main serverless docs site](https://www.serverless.com/framework/docs/providers/aws/events/kafka)
+
+<!-- DOCS-SITE-LINK:END -->
+
+# Kafka
+
+A self-managed Apache Kafka cluster can be used as an event source for AWS Lambda.
+
+## Simple event definition
+
+In the following example, we specify that the `compute` function should be triggered whenever there are new messages available to consume from defined Kafka `topic`.
+
+In order to configure `kafka` event, you have to provide three required properties:
+
+- `accessConfigurations`, which is either secret credentials required to do [SASL_SCRAM auth](https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_scram.html) or basic auth, or this is VPC configuration to allow Lambda to connect to your cluster
+- `topic` to consume messages from.
+- `bootstrapServers` an array of bootstrap server addresses for your Kafka cluster
+
+```yml
+functions:
+  compute:
+    handler: handler.compute
+    events:
+      - kafka:
+          accessConfigurations:
+            saslScram512Auth: arn:aws:secretsmanager:us-east-1:01234567890:secret:MyBrokerSecretName
+          topic: AWSKafkaTopic
+          bootstrapServers:
+            - abc3.xyz.com:9092
+            - abc2.xyz.com:9092
+```
+
+## Using VPC configurations
+
+You can also specify VPC configurations for your event source. The values will be automatically transformed into their corresponding URI values, so it not required to specify the URI prefix. For example, `subnet-0011001100` will be automatically mapped to the value `subnet:subnet-0011001100`.
+
+```yml
+functions:
+  compute:
+    handler: handler.compute
+    events:
+      - kafka:
+          accessConfigurations:
```

Comment:
Which approach is the valid one in the end? I see in the documentation here that `accessConfigurations` are supposed to be an Object, however, in tests and in processing logic it expects it to be an array. Based on recent discussion here https://github.com/serverless/serverless/issues/8718 I got the impression that we could go with object-based approach.

Reply:
Yep, good catch.  I think my documentation is right, but the code may need some fixes.

Reply:
Thanks for clarification :raised_hands: